app/src/main/java/ch/epfl/sdp/musiconnect/GoogleLogin.java

```diff
@@ -0,0 +1,97 @@
+package ch.epfl.sdp.musiconnect;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.util.Log;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import com.google.android.gms.auth.api.signin.GoogleSignIn;
+import com.google.android.gms.auth.api.signin.GoogleSignInAccount;
+import com.google.android.gms.auth.api.signin.GoogleSignInClient;
+import com.google.android.gms.auth.api.signin.GoogleSignInOptions;
+import com.google.android.gms.common.SignInButton;
+import com.google.android.gms.common.api.ApiException;
+import com.google.android.gms.tasks.Task;
+
+import ch.epfl.sdp.R;
+
+public class GoogleLogin extends AppCompatActivity {
+
+    SignInButton signInButton;
+    private GoogleSignInClient mGoogleSignInClient;
+    private int RC_SIGN_IN = 0;
+    private boolean tests = false; // variable telling us if we are running the travis's tests
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_google_login);
+
+        // Set the dimensions of the sign-in button.
+        signInButton = findViewById(R.id.sign_in_button);
+        signInButton.setSize(SignInButton.SIZE_STANDARD);
+        signInButton.setOnClickListener(v -> signIn());
+
+        // Configure sign-in to request the user's ID, email address, and basic
+        // profile. ID and basic profile are included in DEFAULT_SIGN_IN.
+        GoogleSignInOptions gso = new GoogleSignInOptions.Builder(GoogleSignInOptions.DEFAULT_SIGN_IN)
+                .requestEmail()
+                .build();
+
+        // Build a GoogleSignInClient with the options specified by gso.
+        mGoogleSignInClient = GoogleSignIn.getClient(this, gso);
+
+        if(tests)
+            mGoogleSignInClient.signOut(); // used to assume be logged out before each tests
```

Comment:
Not a good practice to have test code into the production application. I would suggest to have a method to logout (that will be use in other places like settings) and call it at the beginning of each test (using @Before statement).